floodsub_test.go

```diff
@@ -1240,7 +1240,7 @@ func TestDedupInboundStreams(t *testing.T) {
 	}
 
 	// Connect the two hosts together
-	connect(t, h2, h1)
+	connect(t, h1, h2)
```

Comment:
This is needed because the go-libp2p dialer now prioritises quic. In quic conn establishment first happens on the dialer side then on the listener side. So we make h2 the dialer here.